### ISSUE TITLE
Masses in R3BRoot / FairRoot

### ISSUE THREAD

**Issue by janmayer** [R3BRootGroup/R3BRoot#338]
I stumbled over a bug with masses in R3BRoot/FairRoot: The MCTrack branch filled during Simulation does not contain the masses for ions anymore, i.e.:

```python
tfile = ROOT.TFile.Open(simfile)
ttree = tfile.Get("evt")
for event in ttree:
    print(event.MCTrack.GetEntries())
    for mct in event.MCTrack:
        print("\t", mct.GetPdgCode(), mct.GetMotherId(), mct.GetMass())
```
```
7
	 1000501310 -1 0.0
	 2112 -1 0.939565
	 1000020040 1 0.0
	 1000020040 1 0.0
	 1000020040 1 0.0
	 2112 1 0.939565
	 2212 1 0.938272
```

I briefly looked when and why that changed, but couldn't find it. I found some [strange logic in FairRoot](https://github.com/FairRootGroup/FairRoot/blob/dev/base/sim/FairPrimaryGenerator.cxx#L278) and [some legacy in R3BMCTrack](https://github.com/R3BRootGroup/R3BRoot/blob/dev/r3bdata/R3BMCTrack.cxx#L140), but oh well.

For the primary particles I could fix that, e.g. in the [Ascii Generator](https://github.com/R3BRootGroup/R3BRoot/blob/dev/r3bgen/R3BAsciiGenerator.cxx#L169), by explicitly passing the mass to [AddTrack](https://github.com/FairRootGroup/FairRoot/blob/33b76b2bbf9fceb482e6f0e163e9fe1871244f54/base/sim/FairPrimaryGenerator.h#L83) but then I 
a) have to re-simulate everything again
b) rely on the values in the input files (which should **not** contain this information, especially not repeated 10000 times)
c) still have no masses for secondary ions (currently not a big problem)
d) have the *wrong* masses from the `TDatabasePdg`, i.e. `0.939 565 00` vs ` 0.939 565 361` in the Input-file vs [the actual value](https://physics.nist.gov/cgi-bin/cuu/Value?mnc2mev) `0.939 565 420 52(54)` vs whatever is actually used in the simulation.

Isn't there some database included one can query easily? In principle, that is just using the existing list of data for all nuclei (and particles) ...

**Comment by janmayer** [comment]
Also, the ion masses in all the 5n and 6n input files are missing or broken:

Example:
```sh
curl -s "https://www.r3broot.gsi.de/data/event_600AMeV_100keV.tar.gz" | tar --strip 1 -C . -xvz
head 127Sn_5n_600AMeV_100keV.dat
```
```
      0      6      0.0      0.0
            -1     50    127     -0.0030777      0.0052615    154.3714142      0.0000000      0.0000000      0.0000000      0.000000
             1      0   2112      0.0009804     -0.0010860      1.2214833      0.0000000      0.0000000      0.0000000      0.939565361
             1      0   2112      0.0059426     -0.0039490      1.2326452      0.0000000      0.0000000      0.0000000      0.939565361
             1      0   2112     -0.0044107      0.0022401      1.2227471      0.0000000      0.0000000      0.0000000      0.939565361
             1      0   2112      0.0028534     -0.0027902      1.2375298      0.0000000      0.0000000      0.0000000      0.939565361
             1      0   2112     -0.0022879      0.0003236      1.2321169      0.0000000      0.0000000      0.0000000      0.939565361
```

**Comment by janmayer** [comment]
Also, the ion masses in the 1n-4n input files are in u, but [FairIon expects GeV/c²](https://github.com/FairRootGroup/FairRoot/blob/dev/base/sim/FairIon.h#L105).

**Comment by kresan** [comment]
@janmayer is it fixed now with #341 ?

**Comment by janmayer** [comment]
> @janmayer is it fixed now with #341 ?

Yes.
